### PR TITLE
Fix #38: Add variable-length data example under scripts

### LINKED ISSUE #38: Add variable-length data example under scripts
## Summary
Add a new `scripts/` example that demonstrates training on trajectories with non-uniform lengths.  Name the example `lti_vlen`.

## Problem
Current examples appear to assume equal-length trajectories. The rough note suggests the `Data` class may already allow variable-length trajectories, but related collation and training paths may not be exercised by any example yet. That leaves a gap in developer-facing coverage and makes it unclear how variable-length data is expected to work in practice.

## Expected outcome
Create a script under `scripts/` that:
- uses a simple 2-state LTI system,
- uses full-state measurements,
- generates or loads multiple trajectories with different lengths,
- follows the existing if-block style used by other script examples such as the `linear_time_invariant` examples,
- runs two optimizer/training variants for comparison: one using NODE and one using the weak-form approach.

## Acceptance criteria
- A new example script exists under `scripts/` for variable-length trajectory data.
- The example clearly constructs a dataset with non-uniform trajectory lengths rather than padding everything to a uniform length up front.
- The example is kept intentionally simple: 2-state LTI with full-state measurement.
- The script includes separate if-block entry points or branches matching the style used in comparable existing examples.
- One branch/config exercises a NODE-based optimizer.
- One branch/config exercises a weak-form optimizer.
- The script is runnable enough for a developer to use as a reference for variable-length data support.
- If variable-length support is incomplete in collation or training, the issue is surfaced explicitly in comments or follow-up notes rather than worked around silently.

## Notes
- Likely reference style: existing examples in or around `scripts/linear_time_invariant`, but the exact file to mirror should be chosen during implementation.
- The rough note implies backend support may need adaptation beyond just adding the example. If so, that should be called out during implementation rather than assumed away.

### PR DESCRIPTION
**Pull request by smanist**
Closes #38

## Summary
This PR was generated by the local AI issue worker.

## Verification
PASS make check
exit code: 0
duration: 4.53s
stdout:
ruff check .
All checks passed!
ruff format . --check
266 files already formatted
pyright --pythonpath "####/miniconda3/bin/python"
0 errors, 0 warnings, 0 informations

stderr:

PASS pytest -m "not slow and not extra_slow"
exit code: 0
duration: 89.99s
stdout:
ript.py:1488: DeprecationWarning: `torch.jit.script` is deprecated. Please switch to `torch.compile` or `torch.export`.
    warnings.warn(

tests/test_assert_krr.py::test_krr
tests/test_assert_krr.py::test_krr
tests/test_assert_krr.py::test_krr
tests/test_assert_krr.py::test_krr
tests/test_assert_krr.py::test_krr
tests/test_assert_krr.py::test_krr
tests/test_assert_krr.py::test_krr
tests/test_assert_krr.py::test_krr
  ####/miniconda3/lib/python3.12/site-packages/torch/_tensor.py:1120: DeprecationWarning: __array_wrap__ must accept context and return_scalar arguments (positionally) in the future. (Deprecated NumPy 2.0)
    return self.reciprocal() * other

tests/test_workflow_sa_lti.py::test_sa[1]
  ####/Repos/dymad-dev/.ai-worktrees/issue-38/src/dymad/training/ls_update.py:368: UserWarning: Casting complex values to real discards the imaginary part (Triggered internally at ####/work/pytorch/pytorch/pytorch/aten/src/ATen/native/Copy.cpp:308.)
    Wt = torch.tensor(W, dtype=dtype, device=device)

-- Docs: https://docs.pytest.org/en/stable/how-to/capture-warnings.html
========== 388 passed, 71 deselected, 29 warnings in 86.93s (0:01:26) ==========

stderr:

## Changed files
- scripts/lti_vlen/lti_vlen.py
- tests/test_workflow_lti_vlen.py

## Agent notes
See diff for implementation details.

### CODE REVIEW

**Comment by smanist** [comment]
More commits done after PR.
